packages/astro/src/jsx/rehype.ts

```diff
@@ -30,7 +30,7 @@ export const rehypeAnalyzeAstroMetadata: RehypePlugin = () => {
 			if (node.type !== 'mdxJsxFlowElement' && node.type !== 'mdxJsxTextElement') return;
 
 			const tagName = node.name;
-			if (!tagName || !isComponent(tagName) || !hasClientDirective(node)) return;
```

Comment:
could removing this condition be why tests are failing? should you instead change the condition to also account for `hasServerDeferDirective` but otherwise leave in the hasClientDirective check?

Reply:
Thanks for reviewing this. Sure I will include `hasServerDeferDirective` in the existing condition.